src/Compilers/CSharp/Portable/Binder/LocalInProgressBinder.cs

```diff
@@ -17,20 +18,26 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal sealed class LocalInProgressBinder : Binder
     {
-        private readonly LocalSymbol _inProgress;
+        public readonly EqualsValueClauseSyntax InitializerSyntax;
+        private LocalSymbol? _localSymbol;
 
-        internal LocalInProgressBinder(LocalSymbol inProgress, Binder next)
+        internal LocalInProgressBinder(EqualsValueClauseSyntax initializerSyntax, Binder next)
             : base(next)
         {
-            _inProgress = inProgress;
+            InitializerSyntax = initializerSyntax;
         }
 
-        internal override LocalSymbol LocalInProgress
+        internal override LocalSymbol? LocalInProgress
         {
             get
             {
-                return _inProgress;
+                return _localSymbol;
```

Comment:
Is it sensible to access this while `_localSymbol` is still `null`? Or does that mean that we forgot to call `SetLocalSymbol` somewhere?

Reply:
It should no longer be `null` by the time and if someone asks. Added an assertion